simple_history/management/commands/clean_duplicate_history.py

```diff
@@ -106,17 +113,29 @@ def _process_instance(self, instance, model, stop_date=None, dry_run=True):
             o_qs = o_qs.filter(history_date__gte=stop_date)
         else:
             extra_one = None
-        with transaction.atomic():
-            # ordering is ('-history_date', '-history_id') so this is ok
-            f1 = o_qs.first()
-            if not f1:
-                return
-
-            for f2 in o_qs[1:]:
-                entries_deleted += self._check_and_delete(f1, f2, dry_run)
-                f1 = f2
-            if extra_one:
-                entries_deleted += self._check_and_delete(f1, extra_one, dry_run)
+
+        # ordering is ('-history_date', '-history_id') so this is ok
+        f1 = o_qs.first()
+        if not f1:
+            return
+
+        batch_count = (len(o_qs) - 1) // batch_size
+        if (len(o_qs) - 1) % batch_size > 0:
+            batch_count += 1
+
+        for batch_index in range(batch_count):
+            with transaction.atomic():
+                batch_start = batch_size * batch_index + 1
+                batch_end = min(batch_start + batch_size, len(o_qs))
+                for i in range(batch_start, batch_end):
+                    f2 = o_qs[i]
+                    entries_deleted += self._check_and_delete(f1, f2, dry_run)
+                    f1 = f2
+                if extra_one:
+                    entries_deleted += self._check_and_delete(f1, extra_one, dry_run)
+                    extra_one = None
+            if batch_index != batch_count - 1:
+                time.sleep(batch_sleep)
```

Comment:
should only call sleep if batch_sleep is not zero, i think `sleep(0)` isn't exactly a no-op and setting batch_sleep to zero should be a viable way to avoid sleeping altogether.

Reply:
Added!